### PR TITLE
Ldld 1

### PR DESCRIPTION
**Pull request by shuangzhou**
This request include impl of 3 simple gradle tasks runQuery, runUpdate and audioChecker to help with the implementation and testing of scripts required in LDLD-1 and LDLD-3 for DCM PI resolving of oral history sound recording records.

### CODE REVIEW

**Comment by shuangzhou** [comment]
@jrixon @terenceingram @scoen @ninhnguyen @yetti @weijunnla @MingtaoSun @snezanam @josephcurtis ?

File: dcm-pi-resolving/src/main/groovy/au/gov/nla/AudioChecker.groovy [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+import java.net.URL;
+import java.nio.file.*
+import joptsimple.internal.Strings;
+import amberdb.AmberSession
+import amberdb.enums.CopyRole
+import amberdb.model.Work
+import amberdb.PIUtil
+
+class AudioUtil {
+    def checkeWorkForAudioDownloadLinks(Work startFromWork, Work currentWork, File log, String appenv) {
```
Comment (scoen) [inline dcm-pi-resolving/src/main/groovy/au/gov/nla/AudioChecker.groovy]: minor spelling: checke

File: dcm-pi-resolving/src/main/groovy/au/gov/nla/AudioChecker.groovy [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        if (currentWork.hasCopyRole(CopyRole.LISTENING_1_COPY)) {
+            def dcmAltPi = ""
+            if (startFromWork.getDcmAltPi() != null && !startFromWork.getDcmAltPi().isEmpty()) {
+                dcmAltPi = " (dcmAltPi: " + Strings.join(startFromWork.getDcmAltPi(), "") + ")"
```
Comment (scoen) [inline dcm-pi-resolving/src/main/groovy/au/gov/nla/AudioChecker.groovy]: just curious as to why the join ?

Comment (scoen) [inline dcm-pi-resolving/src/main/groovy/au/gov/nla/AudioChecker.groovy]: oh, i see - to unwrap from the array

Comment (shuangzhou) [inline dcm-pi-resolving/src/main/groovy/au/gov/nla/AudioChecker.groovy]: @scoen, I was still thinking in java for multi-valued field, good point, I'll update.

File: dcm-pi-resolving/src/main/groovy/au/gov/nla/AudioChecker.groovy (hunk 1 above)
Comment (shuangzhou) [inline dcm-pi-resolving/src/main/groovy/au/gov/nla/AudioChecker.groovy]: @scoen, good point, will update.

File: dcm-pi-resolving/src/main/groovy/au/gov/nla/GroovyCLI.groovy [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    }
+    return statements
+  }
+
+  def greeting(msg, env, url, user, password, fileName, logName) {
+    println msg
+    println "env:         " + env
+    println "amber.url:   " + url
+    println "amber.user:  " + user
+    println "amber.pass:  " + password
```
Comment (scoen) [inline dcm-pi-resolving/src/main/groovy/au/gov/nla/GroovyCLI.groovy]: uh, maybe not ?

Comment (shuangzhou) [inline dcm-pi-resolving/src/main/groovy/au/gov/nla/GroovyCLI.groovy]: @scoen password are now blanked by stars.

File: dcm-pi-resolving/src/main/groovy/au/gov/nla/AudioChecker.groovy (hunk 2 above)
Comment (shuangzhou) [inline dcm-pi-resolving/src/main/groovy/au/gov/nla/AudioChecker.groovy]: @scoen, this has been updated.

File: dcm-pi-resolving/src/main/groovy/au/gov/nla/AudioChecker.groovy (hunk 1 above)
Comment (shuangzhou) [inline dcm-pi-resolving/src/main/groovy/au/gov/nla/AudioChecker.groovy]: @scoen, updated now.

File: dcm-pi-resolving/src/main/groovy/au/gov/nla/AudioChecker.groovy [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+log.delete()
+AudioUtil audioUtil = new AudioUtil()
+AmberSession session = new GroovyCLI().createAmberDb(url, user, password, path).begin()
+File dataFile = new File(System.getProperty("file"))
+
+if ("checkForListeningSessions".equals(op)) {
+    dataFile.eachLine { line ->
+	    Work work = session.findWork(Long.parseLong(audioUtil.resolvePI(line, appenv)))
+	    audioUtil.navigateWorksForListeningSession(work, work, log)
+	    if (!log.exists()) {
```
Comment (scoen) [inline dcm-pi-resolving/src/main/groovy/au/gov/nla/AudioChecker.groovy]: just my confusion maybe, but are you only writing to the log if it doesn't exist ?

File: dcm-pi-resolving/src/main/groovy/au/gov/nla/GroovyCLI.groovy [hunk 5]
```diff
@@ -0,0 +1,52 @@
+package au.gov.nla
+
+import com.mysql.jdbc.*
+import com.mysql.jdbc.jdbc2.optional.*
+import java.nio.file.*
+import org.skife.jdbi.v2.DBI
+import amberdb.AmberDb
+
+class GroovyCLI {
```
Comment (scoen) [inline dcm-pi-resolving/src/main/groovy/au/gov/nla/GroovyCLI.groovy]: Just a naming thing: this doesn't seem to be a Command Line Interface (CLI).

File: dcm-pi-resolving/src/main/groovy/au/gov/nla/AudioChecker.groovy (hunk 4 above)
Comment (shuangzhou) [inline dcm-pi-resolving/src/main/groovy/au/gov/nla/AudioChecker.groovy]: @scoen, the log is also written to in the navigateWorksForListeningSession.  I also want to write a log entry if there's no work found without a listening session, that's the reason for checking whether the log file exists.  I should move the check for be outside the for each loop though, I'll update now.

File: dcm-pi-resolving/src/main/groovy/au/gov/nla/GroovyCLI.groovy (hunk 5 above)
Comment (shuangzhou) [inline dcm-pi-resolving/src/main/groovy/au/gov/nla/GroovyCLI.groovy]: @scoen, I'll rename this to indicate it's a utility class with a set of common methods shared between the dcm pi resolving scripts.

Comment (scoen) [inline dcm-pi-resolving/src/main/groovy/au/gov/nla/GroovyCLI.groovy]: sounds good

**Comment by shuangzhou** [comment]
@scoen the GroovyCLI are now renamed to GroovyUtil.

**Pull request merged**